homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,277 @@
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.typing import HomeAssistantType
+import homeassistant.helpers.config_validation as cv
+from . import LyricDeviceEntity
+from .const import DATA_LYRIC_CLIENT, DATA_LYRIC_DEVICES, DOMAIN, SERVICE_HOLD_TIME
+
+_LOGGER = logging.getLogger(__name__)
+
+PRESET_NO_HOLD = "NoHold"
+PRESET_HOLD_UNTIL = "HoldUntil"
+PRESET_PERMANENT_HOLD = "PermanentHold"
+PRESET_TEMPORARY_HOLD = "TemporaryHold"
+PRESET_VACATION_HOLD = "VacationHold"
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+HOLD_PERIOD_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.comp_entity_ids,
+        vol.Required(ATTR_TIME): cv.string,
+    }
+)
+
+LYRIC_HVAC_MODES = {
+    HVAC_MODE_OFF: "OFF",
+    HVAC_MODE_HEAT: "HEAT",
+    HVAC_MODE_COOL: "COOL",
+    HVAC_MODE_HEAT_COOL: "HEAT_COOL",
+}
+
+HVAC_MODES = {
+    "OFF": HVAC_MODE_OFF,
+    "HEAT": HVAC_MODE_HEAT,
+    "COOL": HVAC_MODE_COOL,
+    "HEAT_COOL": HVAC_MODE_HEAT_COOL,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up Lyric thermostat based on a config entry."""
+    lyric = hass.data[DOMAIN][DATA_LYRIC_CLIENT]
+
+    try:
+        devices = lyric.devices()
+    except ConnectionError as exception:
+        raise PlatformNotReady from exception
+
+    hass.data[DOMAIN][DATA_LYRIC_DEVICES] = devices
+
+    temp_unit = hass.config.units.temperature_unit
+    entities = [
+        LyricThermostat(device, location, temp_unit) for location, device in devices
+    ]
+
+    async_add_entities(entities, True)
+
+    async def hold_time_service(service) -> None:
+        """Set the time to hold until."""
+        entity_ids = service.data[ATTR_ENTITY_ID]
+        time = service.data[ATTR_TIME]
+
+        _LOGGER.debug("hold_time_service: %s; %s", entity_ids, time)
+
+        if entity_ids == "all":
+            target_thermostats = entities
+        else:
+            target_thermostats = [
+                entity for entity in entities if entity.entity_id in entity_ids
+            ]
+
+        _LOGGER.debug("target_thermostats: %s", target_thermostats)
+
+        for thermostat in target_thermostats:
+            await asyncio.wait({thermostat.async_set_preset_period(time)})
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_HOLD_TIME, hold_time_service, schema=HOLD_PERIOD_SCHEMA
+    )
+
+
+class LyricThermostat(LyricDeviceEntity, ClimateDevice):
+    """Representation of a Lyric thermostat."""
+
+    def __init__(self, device, location, temp_unit) -> None:
+        """Initialize the thermostat."""
+        unique_id = "{}_climate".format(device.macID)
+        name = "{}".format(device.name)
+
+        self._unit = temp_unit
+
+        # Setup supported hvac modes
+        self._hvac_modes = [HVAC_MODE_OFF]
```

Comment:
let's append `HVAC_MODE_OFF` at the end. Some UIs might use the sort mode that is in the state machine and it's weird if `off` is the first option instead of last.